adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -88,26 +88,35 @@ - (void)renderNativeAdForAdConfiguration:
                  strongSelf->_nativeAd = nativeAd;
                  strongSelf->_nativeAd.delegate = strongSelf;
                  strongSelf->_nativeAd.rootViewController = adConfiguration.topViewController;
-
-                 if (strongSelf->_loadCompletionHandler) {
-                   id<GADMediationNativeAdEventDelegate> delegate =
-                       strongSelf->_loadCompletionHandler(strongSelf, nil);
-                   strongSelf->_delegate = delegate;
-                 }
+      
+                 [strongSelf loadRequireData];
                }];
 }
 
-#pragma mark - GADMediationNativeAd
-
-- (GADNativeAdImage *)icon {
-  if (!_icon) {
-    if (_nativeAd.data.icon && _nativeAd.data.icon.imageURL != nil) {
-      _icon = [self imageWithUrlString:_nativeAd.data.icon.imageURL];
-    }
-  }
-  return _icon;
+- (void)loadRequireData {
+    NSString *urlString = _nativeAd.data.icon.imageURL;
+    dispatch_async(dispatch_get_global_queue(0, 0), ^{
+        NSURL *url = [NSURL URLWithString:urlString];
+        NSData *data = [NSData dataWithContentsOfURL:url];
+        GADPangleRTBNativeRenderer *__weak weakSelf = self;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            GADPangleRTBNativeRenderer *strongSelf = weakSelf;
+            if (!strongSelf) {
+               return;
+            }
```

Comment:
Do we really need the `weakSelf` and `strongSelf` implementations here? We already do that above.

Reply:
I will remove it.